File: 0808-25-2379-minimum-recolors-to-get-k-consecutive-black-blocks/0808-25-2379-minimum-recolors-to-get-k-consecutive-black-blocks.cpp
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        int w = count(blocks.begin(),blocks.begin()+k,'W'),ans = w;
        for(int i = k ; i < blocks.size();i++)
            w+= (blocks[i]=='W') - (blocks[i-k]=='W'), ans=min(ans,w);
        return ans;
    }
};

File: 0808-25-2379-minimum-recolors-to-get-k-consecutive-black-blocks/0808-25-2379-minimum-recolors-to-get-k-consecutive-black-blocks.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
        int w = 0;
        for (int i = 0; i < k; i++) {
            if (blocks.charAt(i) == 'W') w++;
        }
        int ans = w;
        for (int i = k; i < blocks.length(); i++) {
            if (blocks.charAt(i - k) == 'W') w--;
            if (blocks.charAt(i) == 'W') w++;
            ans = Math.min(ans, w);
        }
        return ans;
    }
}